### PR TITLE
Use struct handlers to capture dependencies

### PR DESCRIPTION
**Pull request by jim**
## Description

@macrael and I have been thinking about how to make the dependencies for our handlers more explicit and remove the use of globals as much as possible. This facilitates testing and makes the code easier to reason about.

## Reviewer Notes
The main pattern is to change a handler definition that was like this:

```golang
// pkg/handlers/form1299s.go
func ShowForm1299Handler(params form1299op.ShowForm1299Params) middleware.Responder {
	// use global dbConnection to access database and zap.L() for logging
}

// cmd/webserver/main.go
internalAPI.Form1299sShowForm1299Handler = form1299op.ShowForm1299HandlerFunc(handlers.ShowForm1299Handler)
```

into:

```golang
// pkg/handlers/form1299s.go
type ShowForm1299Handler struct {
	db     *pop.Connection
	logger *zap.Logger
}

func NewShowForm1299Handler(db *pop.Connection, logger *zap.Logger) ShowForm1299Handler {
	return ShowForm1299Handler{
		db:     db,
		logger: logger,
	}
}
func (h ShowForm1299Handler) Handle(params form1299op.ShowForm1299Params) middleware.Responder {
	// use h.db to access database and h.logger for logging
}

// cmd/webserver/main.go
internalAPI.Form1299sShowForm1299Handler = handlers.NewShowForm1299Handler(dbConnection, logger)
```

The database connection and logger are still instantiated within `cmd/webserver/main.go`, but the resulting values are passed into the handlers explicitly. There is a bit more boilerplate, but the benefit is that a handler's dependencies are explicitly defined within the handler struct.

## Verification Steps

* [x] All tests pass.

## References

* [this article](http://blog.questionable.services/article/custom-handlers-avoiding-globals/) explains more about the approach used.

### CODE REVIEW

File: pkg/handlers/handlers_test.go [hunk 1]
```diff
@@ -2,38 +2,36 @@ package handlers
...
-
-func setupDBConnection() {
 	configLocation := "../../config"
 	pop.AddLookupPaths(configLocation)
-	dbConnection, err := pop.Connect("test")
+	db, err := pop.Connect("test")
 	if err != nil {
 		log.Panic(err)
 	}
+	Init(db)
```
Comment (macrael) [inline pkg/handlers/handlers_test.go]: Do we still need this? If all the handlers have been converted then I think we should be able to skip this from now on.

File: cmd/webserver/main.go [hunk 2]
```diff
@@ -83,14 +80,14 @@ func main() {
```
Comment (macrael) [inline cmd/webserver/main.go]: We should be able to get rid of the handlers.Init call above as well, and delete that method and package variable from handlers.go

Comment (jim) [reply cmd/webserver/main.go]: Good call. Done.

**Comment by jim** [comment]
@macrael Want to give this another once-over? I think we're ready to remove the `WIP` and get it merged in.

**Pull request merged**